### PR TITLE
Utilities: Fix top utility not calling exit() on SIGINT

### LINKED ISSUE #16306: Terminal(?): Pressing Ctrl+C in `top` exits the Terminal app
To reproduce, run `top` and press Ctrl+C. The terminal app should not exit, but it does :thinking:

### PR DESCRIPTION
**Pull request by HawDevelopment**
Fixes #16306.

### CODE REVIEW

File: Userland/Utilities/top.cpp [hunk 1]
```diff
@@ -203,11 +203,13 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
-    signal(SIGWINCH, [](int) {
+    TRY(Core::System::signal(SIGWINCH, [](int) {
         g_window_size_changed = true;
-    });
+    }));
+    TRY(Core::System::signal(SIGINT, [](int) {
+        exit(0);
+    }));
 
-    TRY(Core::System::pledge("stdio rpath tty"));
```
Comment (trflynn89) [inline Userland/Utilities/top.cpp:210]: Why is this removed?

Comment (HawDevelopment) [reply Userland/Utilities/top.cpp:210]: It was overwriting the past pledge at the top of the function.

**Review by MacDue — COMMENTED** [review]
Little confused, why should this change be made if you say there's already another PR (https://github.com/SerenityOS/serenity/pull/16375) that solves this more "long-term"?

**Comment by HawDevelopment** [comment]
> Little confused, why should this change be made if you say there's already another PR (#16375) that solves this more "long-term"?

This change will solve the terminal 'crashing' abnormally. #16375 will fix a shell problem, that won't be triggered with this pr, but will stop it from happening in the future. I'll edit the pr to better state this.

**Comment by MacDue** [comment]
Both your PR and https://github.com/SerenityOS/serenity/pull/16375 say they fix https://github.com/SerenityOS/serenity/issues/16306, which is a little confusing. 
Are both these changes needed to fix the issue? Or do both PRs fix the issue in different ways?

**Comment by HawDevelopment** [comment]
> Both your PR and #16375 say they fix #16306, which is a little confusing. Are both these changes needed to fix the issue? Or do both PRs fix the issue in different ways?

This pr will fix the terminal exiting abnormally. The other pr won't fix that, instead a problem related to the shell and reading console characters.

Both problems are caused be the top atexit handler not being executed on a SIGINT signal. That's what this pr fixes.

**Comment by kleinesfilmroellchen** [comment]
I assume that both PRs are going to be useful, even if the other PR hides the bug that this PR fixes.

File: Userland/Utilities/top.cpp (hunk 1 above)
Comment (HawDevelopment) [reply Userland/Utilities/top.cpp:210]: Sorry, I got an error when testing about a missing pledge. This line has been re-added.

**Pull request merged**